### PR TITLE
IDE-609 ECM attributes need to be treated as ESDL attributes

### PR DESCRIPTION
**Pull request by dehilsterlexis**
Fixed IDE-609

### CODE REVIEW

**Comment by richardkchapman** [comment]
@GordonSmith Please review

File: comms/AttributeType.cpp [hunk 1]
```diff
@@ -207,10 +205,8 @@ class CAttributeType : public IAttributeType, public CUnknown
     {
         if (boost::algorithm::equals(m_repositoryCode, ATTRIBUTE_TYPE_ECL))
             m_defaultText = (boost::_tformat(_T("EXPORT %1% := 'todo';")) % attrLabel.c_str()).str();
-        else if (boost::algorithm::equals(m_repositoryCode, ATTRIBUTE_TYPE_ESDL))
-            m_defaultText = (boost::_tformat(_T("ESDL:  %s;")) % attrLabel.c_str()).str();
-        else if (boost::algorithm::equals(m_repositoryCode, ATTRIBUTE_TYPE_ECM))
-            m_defaultText = (boost::_tformat(_T("ECM:  %s;")) % attrLabel.c_str()).str();
+        else if (boost::algorithm::equals(m_repositoryCode, ATTRIBUTE_TYPE_ECM) || boost::algorithm::equals(m_repositoryCode, ATTRIBUTE_TYPE_ESDL))
+            m_defaultText = (boost::_tformat(_T("ECM (ESDL):  %s;")) % attrLabel.c_str()).str();
```
Comment (GordonSmith) [inline comms/AttributeType.cpp]: "ESDL (ECM)..." for consistency?

**Comment by dehilsterlexis** [comment]
@GordonSmith back to you.

**Comment by richardkchapman** [comment]
Should this target 6.4.0?

**Comment by GordonSmith** [comment]
Yup should be targeting 6.4

**Comment by richardkchapman** [comment]
@dehilsterlexis Please retarget

**Comment by dehilsterlexis** [comment]
I don't think I did this correctly.

**Comment by richardkchapman** [comment]
Looks ok, but perhaps @GordonSmith can sanity-check?

File: wlib/LangRef.cpp [hunk 2]
```diff
@@ -473,7 +473,26 @@ class CLangRef : public ILangRef, public CUnknown
...
+				return SCLEX_KEL;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_DUD))
+			{
+				return SCLEX_DUD;
+			}
+			else if (boost::algorithm::equals(m_elementType, ATTRIBUTE_TYPE_GENERAL))
+			{
+				return SCLEX_GENERAL;
+			}
```
Comment (GordonSmith) [inline wlib/LangRef.cpp]: Not sure the other lang types should be in this PR - but its ok to leave them there.

File: wlib/PreferenceDlg.cpp [hunk 3]
```diff
@@ -1158,6 +1158,89 @@ class CPrefColorDlg : public CDialogImpl<CPrefColorDlg>,
...
+		int n;
+		n = m_comboElement.GetLBTextLen(row);
+		m_comboElement.GetLBText(row, str1.GetBuffer(n));
+		str1.ReleaseBuffer();
+		str1.MakeLower();
+
+		m_currentElementStr = str1;
+
+		return str1;
+	}
```
Comment (GordonSmith) [inline wlib/PreferenceDlg.cpp]: You can common up some of the above code?

File: wlib/PreferenceDlg.cpp [hunk 4]
```diff
@@ -1306,6 +1389,163 @@ class CPrefColorDlg : public CDialogImpl<CPrefColorDlg>,
...
+		std::_tstring textSample = _T("");
+		bool filled = false;
+
+		if (m_langRef)
+		{
+			//m_langRef->loadSamples();
+			textSample = m_langRef->GetSample();
+			filled = true;
+
+			if (textSample == _T(""))
```
Comment (GordonSmith) [inline wlib/PreferenceDlg.cpp]: textSample.empty() might be preferable (minor)

File: wlib/PreferenceDlg.cpp [hunk 5]
```diff
@@ -1306,6 +1389,163 @@ class CPrefColorDlg : public CDialogImpl<CPrefColorDlg>,
...
+						_T("   string22 s1 := '123';")						_T("\r\n")
+						_T("   integer4 i1 := 123;")						_T("\r\n")
+						_T("  end;")										_T("\r\n")
+						_T("#option('tmp', true);")							_T("\r\n")
+						_T("d := dataset('tmp::qb', r, thor);")				_T("\r\n")
+						_T("output(d);")									_T("\r\n")
+						_T("Compare:  Added")								_T("\r\n")
+						_T("Compare:  Deleted")								_T("\r\n")
+						_T("Compare:  Changed")								_T("\r\n")
+						_T("Compare:  Moved");
```
Comment (GordonSmith) [inline wlib/PreferenceDlg.cpp]: Your \r\n are all over the place - you could just move them inside the other strings?

**Pull request merged**